File: Stack/Stack.java
import java.util.Scanner;

class node
{
    public int data;
    public node next;
}

class StackX
{
    private node first;
    private int iCount;

    
    public StackX()
    {
        this.first = null;
        this.iCount = 0;
    }

    public void push(int no)
    {
        node newn = new node();

        newn.data = no;
        newn.next = null;

        newn.next = first;
        first = newn;

        iCount++;
    }

    public int pop()
    {
        node temp = first;

        if(this.first == null)
        {
            System.out.println("Unable to pop as stack is empty");
            return -1;
        }
        else
        {
            first = first.next;
            iCount--;
            return temp.data;
        }
    }

    public void Display()
    {
        node temp = null;

        temp = this.first;

        while(temp != null)
        {
            System.out.println("| "+temp.data+" |");
            temp = temp.next;
        }
    }

    public int Count()
    {
        return iCount;
    }
}

class Stack
{
    public static void main(String A[])
    {
        StackX sobj = new StackX();
        Scanner sobj1 = new Scanner(System.in);
        int iRet = 0, iValue = 0, iChoice = 0;

        System.out.print("-------------------------------------------------------------------\n");
        System.out.print("---------------------------------Stack------------------------------\n");
        System.out.print("-------------------------------------------------------------------\n\n");

        while(true)
        {
            System.out.print("-------------------------------------------------------------------\n");
            System.out.print("----------------------Please Select the option----------------------\n");
            System.out.print("-------------------------------------------------------------------\n");

            System.out.print("1 : Insert new node in the stack\n");
            System.out.print("2 : Pop Element from the stack\n");
            System.out.print("3 : Display all elements from the Stack\n");
            System.out.print("4 : Count all the elements from the stack\n");
            System.out.print("0 : Terminate the position\n");
            System.out.print("-------------------------------------------------------------------\n");

            iChoice = sobj1.nextInt();

            if(iChoice == 1)
            {
                System.out.print("Enter the data that you want to insert : \n");
                iValue = sobj1.nextInt();
                sobj.push(iValue);
            }
            else if(iChoice == 2)
            {
                System.out.print("Remove the element from the stack\n");
                iRet = sobj.pop();
                System.out.println("Removed element from the stack is : \n"+iRet);
            }
            else if(iChoice == 3)
            {
                sobj.Display();
            }
            else if(iChoice == 4)
            {
                iRet = sobj.Count();
                System.out.print("The number of elements in the stack are : "+iRet);
            }
            else if(iChoice == 0)
            {
                System.out.print("Thank you for using our application\n");
                break;
            }
            else
            {
                System.out.print("Invalid choice\n");
            }
            System.out.print("\n-------------------------------------------------------------------\n");
        }
    }
}

File: Stack/Stack.c
#include<stdio.h>
#include<stdlib.h>

struct node 
{
    int data;
    struct node *next;
};

typedef struct node NODE;
typedef struct node *PNODE;
typedef struct node **PPNODE;

void push(PPNODE first, int no)
{
    PNODE newn = NULL;

    newn = (PNODE)malloc(sizeof(NODE));

    newn->data = no;
    newn->next = NULL;

    newn->next = *first;
    *first = newn;
}

void Display(PNODE first)
{
    while(first != NULL)
    {
        printf("| %d | \n",first->data);
        first = first -> next;
    }
    printf("\n");
}

int pop(PPNODE first)
{
    PNODE temp = *first;

    if(*first == NULL)
    {
        printf("Unable to pop as stack is empty\n");
        return -1;
    }
    else
    {
        *first = (*first)->next;
        return temp->data;
    }
}

int Count(PNODE first)
{
    int iCount = 0;

    while(first != NULL)
    {
        iCount++;
        first = first -> next;
    }
    return iCount;
}

int main()
{
    PNODE head = NULL;
    int iChoice = 0, iValue = 0, iRet = 0;

    printf("-------------------------------------------------------------------\n");
    printf("---------------------------------Stack-----------------------------\n");
    printf("-------------------------------------------------------------------\n\n");

    while(1)
    {
        printf("-------------------------------------------------------------------\n");
        printf("----------------------Please Select the option----------------------\n");
        printf("-------------------------------------------------------------------\n");

        printf("1 : Insert new node in the Stack\n");
        printf("2 : Remove element from the stack\n");
        printf("3 : Display all elements from the stack\n");
        printf("4 : Count all the elements from the stack\n");
        printf("0 : Terminate the position\n");
        printf("-------------------------------------------------------------------\n");

        scanf("%d",&iChoice);

        if(iChoice == 1)
        {
            printf("Enter the data that you want to insert : \n");
            scanf("%d",&iValue);
            push(&head,iValue);
        }
        else if(iChoice == 2)
        {
            printf("Remove the element from the stack\n");
            iRet = pop(&head);
            printf("Removed element from the stack is : %d\n",iRet);
        }
        else if(iChoice == 3)
        {
            Display(head);
        }
        else if(iChoice == 4)
        {
            iRet = Count(head);
            printf("The number of elements in the stack are : %d\n",iRet);
        }
        else if(iChoice == 0)
        {
            printf("Thank you for using our application\n");
            break;
        }
        else
        {
            printf("Invalid choice\n");
        }
        printf("\n-------------------------------------------------------------------\n");
    }

    return 0;
}